java/code/src/com/redhat/rhn/manager/action/ActionChainManager.java

```diff
@@ -309,15 +309,16 @@ private static Action schedulePackageAction(User user, List<Map<String, Long>> p
      * Schedules the application of states to minions
      * @param user the requesting user
      * @param sids list of system IDs
+     * @param stateNames list of State Names. Apply Highstate when empty.
      * @param test for test mode
      * @param earliest earliest execution date
      * @param actionChain Action Chain instance
      * @return a set of Actions
      * @throws TaskomaticApiException if there was a Taskomatic error
      * (typically: Taskomatic is down)
      */
-    public static Set<Action> scheduleApplyStates(User user, List<Long> sids, Optional<Boolean> test,
-                                                   Date earliest, ActionChain actionChain)
+    public static Set<Action> scheduleApplyStates(User user, List<Long> sids, Optional<List<String>> stateNames,
+            Optional<Boolean> test, Date earliest, ActionChain actionChain)
```

Comment:
if passing false for test is equivalent to empty and empty list of stateNames is equivalent to empty optional i would just get rid of the optionals here.